sdk/inc/azure/core/az_mqtt5_rpc_client.h

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition of #az_mqtt5_rpc_client. You use the RPC client to send commands.
+ *
+ * @note The state diagram for this HFSM is in sdk/docs/core/rpc_client.puml
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_MQTT5_RPC_CLIENT_H
+#define _az_MQTT5_RPC_CLIENT_H
+
+#include <azure/core/az_mqtt5_connection.h>
+#include <azure/core/az_mqtt5_rpc.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+// ~~~~~~~~~~~~~~~~~~~~ Codec Public API ~~~~~~~~~~~~~~~~~
+/**
+ * @brief The MQTT5 RPC Client.
+ *
+ */
+typedef struct az_mqtt5_rpc_client az_mqtt5_rpc_client;
+
+/**
+ * @brief MQTT5 RPC Client options.
+ *
+ */
+typedef struct
+{
+  /**
+   * @brief QOS to use for subscribing (MUST be QOS 1)
+   */
+  int8_t subscribe_qos;
+  /**
+   * @brief QOS to use for sending requests (MUST be QOS 1)
+   */
+  int8_t request_qos;
+  /**
+   * @brief timeout in seconds for subscribing (must be > 0)
+   */
+  uint32_t subscribe_timeout_in_seconds;
+
+} az_mqtt5_rpc_client_options;
+
+/**
+ * @brief The MQTT5 RPC Client.
+ *
+ */
+struct az_mqtt5_rpc_client
+{
+  az_span client_id;
+  az_span model_id;
+  az_span executor_client_id;
+  az_span command_name;
+  az_span response_topic;
+  az_span request_topic;
+  /**
+   * @brief Options for the MQTT5 RPC Client.
+   */
+  az_mqtt5_rpc_client_options options;
+};
+
+AZ_NODISCARD az_result
+az_rpc_client_get_response_topic(az_mqtt5_rpc_client* client, az_span* out_response_topic);
```

Comment:
az_span should never be passed as pointers. Use az_span on the stack (copy) instead.